http/http-api/src/main/java/com/okta/commons/http/QueryString.java

```diff
@@ -35,21 +36,30 @@ public QueryString(){}
 
     public QueryString(Map<String,?> source) {
         super();
+
         if (!Collections.isEmpty(source)) {
-            for(Map.Entry<String,?> entry : source.entrySet()) {
-                String key = entry.getKey();
-                Object value = entry.getValue();
-                String sValue = value != null ? String.valueOf(value) : null;
-                put(key, sValue);
+            for (Map.Entry<String,?> entry : source.entrySet()) {
+                put(entry.getKey(), getFormattedValue(entry.getValue()));
             }
         }
     }
 
     public String put(String key, Object value) {
+        return super.put(key, getFormattedValue(value));
+    }
+
+    private String getFormattedValue(Object value) {
+        String result = null;
+
         if (value != null) {
```

Comment:
Is this null check lost on the `put`? For example (if i'm reading the code correctly), the old behavior would not add a null value, but the new one will?

Reply:
Good catch! updated code to prevent this.